lang/cs/Org.Apache.REEF.Tests/Functional/Failure/EvaluatorWithRunningTaskImmediatePoisonTest.cs

```diff
@@ -0,0 +1,138 @@
+
+            // verify that no unexpected events happened
+            ValidateMessageSuccessfullyLoggedForDriver(BasePoisonedEvaluatorDriver.UnexpectedClosedContext, testFolder, 0);
+            ValidateMessageSuccessfullyLoggedForDriver(BasePoisonedEvaluatorDriver.UnexpectedFailedContext, testFolder, 0);
+            ValidateMessageSuccessfullyLoggedForDriver(BasePoisonedEvaluatorDriver.UnexpectedCompletedTask, testFolder, 0);
+            ValidateMessageSuccessfullyLoggedForDriver(BasePoisonedEvaluatorDriver.UnexpectedFailedTask, testFolder, 0);
+            CleanUp(testFolder);
+        }
+
+        private static IConfiguration DriverConfigurations()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextClosed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Build();
+        }
+
+        private sealed class PoisonedEvaluatorDriver :
+            BasePoisonedEvaluatorDriver,
+            IObserver<IFailedEvaluator>,
+            IObserver<IActiveContext>
+        {
+            [Inject]
+            private PoisonedEvaluatorDriver(IEvaluatorRequestor requestor) : base(requestor)
+            {
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                var taskConfig = TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, TaskId)
+                    .Set(TaskConfiguration.Task, GenericType<SleepTask>.Class)
+                    .Set(TaskConfiguration.OnClose, GenericType<SleepTask>.Class)
+                    .Set(TaskConfiguration.OnTaskStart, GenericType<PoisonedEventHandler<ITaskStart>>.Class)
+                    .Build();
+
+                var poisonConfig = TangFactory.GetTang().NewConfigurationBuilder()
+                    .BindIntNamedParam<CrashTimeout>("0")
+                    .BindIntNamedParam<CrashMinDelay>("0")
+                    .BindNamedParameter<CrashProbability, double>(GenericType<CrashProbability>.Class, "1.0")
+                    .Build();
+
+                value.SubmitTask(Configurations.Merge(taskConfig, poisonConfig));
+            }
+
+            public void OnNext(IFailedEvaluator value)
+            {
```

Comment:
No. I'm thinking maybe I should make one of them base and inherit the other from it, so that they don't drift apart over time?